src/test/java/dev/sigstore/testing/FulcioWrapper.java

```diff
@@ -17,22 +17,56 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import org.junit.jupiter.api.extension.*;
 
 /**
  * A test fixture to start fulcio from an executable on the system path. This requires fulcio to be
- * installed, do something like go install github.com/sigstore/fulcio
+ * installed, do something like go install github.com/sigstore/fulcio.
+ *
+ * <p>It will pick up config for a ctLog (via FakeCTLogServer) using the per test storage item
+ * CTLOGURL. It will pick up an issuer for fulcio (via MockOAuth2ServerExtension) using the per test
+ * storage item MOCK_OAUTH_ISSUER.
  */
-public class FulcioWrapper {
+public class FulcioWrapper implements BeforeEachCallback, AfterEachCallback, ParameterResolver {
+
+  private Process fulcioProcess;
+  private Path fulcioLog;
+  private Path fulcioConfig;
 
-  private final Process fulcioProcess;
+  public URI getURI() {
+    return URI.create("http://localhost:5555");
+  }
 
-  private FulcioWrapper(Process p) {
-    this.fulcioProcess = p;
+  private Path createConfig(String issuer) throws IOException {
+    fulcioConfig = Files.createTempFile("fulcio-config", ".json");
+    Files.writeString(
+        fulcioConfig,
+        String.format(
+            "{\"OIDCIssuers\":{ \"%s\": { \"IssuerURL\": \"%s\", \"ClientID\": \"sigstore\", \"Type\": \"email\"}}}",
+            issuer, issuer));
+    return fulcioConfig;
   }
 
-  public static FulcioWrapper startNewServer(Path config, String ctLogUrl)
-      throws IOException, InterruptedException {
+  @Override
+  public void afterEach(ExtensionContext context) throws Exception {
+    // forcible kill fulcio and all it's subprocesses
+    fulcioProcess.destroyForcibly();
+    fulcioProcess.waitFor();
+    System.out.println(Files.readString(fulcioLog));
+    Files.deleteIfExists(fulcioLog);
+    Files.deleteIfExists(fulcioConfig);
```

Comment:
do you get the channel file showing up in your directory tree too? @fulcio-legacy-grpc-socket?

Reply:
no, I've never seen that, maybe the shutdown on mac is a bit weird?